### PR TITLE
issue-2559: ShardBalancer: blocks vs bytes fix + service_ut_sharding testcase

### PR DESCRIPTION
**Pull request by qkrorlqr**
#2559

### CODE REVIEW

**Review by yegorskii — APPROVED** [review]

File: cloud/filestore/libs/storage/service/service_ut_sharding.cpp [hunk 1]
```diff
@@ -4019,6 +4019,133 @@ Y_UNIT_TEST_SUITE(TStorageServiceShardingTest)
...
+            const auto nodeId = createHandleResponse.GetNodeAttr().GetId();
+            shards.insert(ExtractShardNo(nodeId));
+            nodes.push_back(nodeId);
+
+            const auto handleId = createHandleResponse.GetHandle();
+            handles.push_back(handleId);
+        }
+
+        UNIT_ASSERT_VALUES_EQUAL(5, shards.size());
+
```
Comment (budevg) [inline cloud/filestore/libs/storage/service/service_ut_sharding.cpp:4084]: let's add some comments to the test to make it more understandable
such as 
// filled 3 out of 5 shards

File: cloud/filestore/libs/storage/service/service_ut_sharding.cpp [hunk 2]
```diff
@@ -4019,6 +4019,133 @@ Y_UNIT_TEST_SUITE(TStorageServiceShardingTest)
...
+                TCreateHandleArgs::CREATE)->Record;
+
+            const auto nodeId = createHandleResponse.GetNodeAttr().GetId();
+            shards.insert(ExtractShardNo(nodeId));
+            nodes.push_back(nodeId);
+
+            const auto handleId = createHandleResponse.GetHandle();
+            handles.push_back(handleId);
+        }
+
```
Comment (budevg) [inline cloud/filestore/libs/storage/service/service_ut_sharding.cpp:4139]: let's add comments for validation
// files should be balanced between remaining empty two shards

File: cloud/filestore/libs/storage/service/service_ut_sharding.cpp [hunk 3]
```diff
@@ -4019,6 +4019,133 @@ Y_UNIT_TEST_SUITE(TStorageServiceShardingTest)
...
+            auto createHandleResponse = service.CreateHandle(
+                headers,
+                fsId,
+                RootNodeId,
+                Sprintf("file%u", 5 + i),
+                TCreateHandleArgs::CREATE)->Record;
+
+            const auto nodeId = createHandleResponse.GetNodeAttr().GetId();
+            shards.insert(ExtractShardNo(nodeId));
+            nodes.push_back(nodeId);
```
Comment (budevg) [inline cloud/filestore/libs/storage/service/service_ut_sharding.cpp:4134]: do you want to test that you are getting different shard each time to validate that allocation rotates across available shards

File: cloud/filestore/libs/storage/tablet/model/shard_balancer.h [hunk 4]
```diff
@@ -36,15 +36,19 @@ class TShardBalancer
     };
 
 private:
+    ui32 BlockSize;
```
[the remark this answers is not in the document]
Comment (budevg) [reply cloud/filestore/libs/storage/tablet/model/shard_balancer.h:39]: will be resolved in next pr

File: cloud/filestore/libs/storage/service/service_ut_sharding.cpp (hunk 1 above)
Comment (budevg) [reply cloud/filestore/libs/storage/service/service_ut_sharding.cpp:4084]: will be resolved in next pr

File: cloud/filestore/libs/storage/service/service_ut_sharding.cpp (hunk 2 above)
Comment (budevg) [reply cloud/filestore/libs/storage/service/service_ut_sharding.cpp:4139]: will be resolved in next pr

File: cloud/filestore/libs/storage/service/service_ut_sharding.cpp (hunk 3 above)
Comment (budevg) [reply cloud/filestore/libs/storage/service/service_ut_sharding.cpp:4134]: will be resolved in next pr

**Review by budevg — APPROVED** [review]

**Pull request merged**